docs/apis/programming_guide.md

```diff
@@ -394,26 +382,66 @@ def write(outputFormat: FileOutputFormat[T],
     writeMode: WriteMode = WriteMode.NO_OVERWRITE)
 
 def print()
-{% endhighlight %}
 
-The last method is only useful for developing/debugging on a local machine,
-it will output the contents of the DataSet to standard output. (Note that in
-a cluster, the result goes to the standard out stream of the cluster nodes and ends
-up in the *.out* files of the workers).
-The first two do as the name suggests, the third one can be used to specify a
-custom data output format. Please refer
-to [Data Sinks](#data-sinks) for more information on writing to files and also
-about custom data output formats.
-
-Once you specified the complete program you need to call `execute` on
-the `ExecutionEnvironment`. This will either execute on your local
-machine or submit your program for execution on a cluster, depending on
-how you created the execution environment.
+def collect()
+{% endhighlight %}
 
 </div>
 </div>
 
 
+The first two methods (`writeAsText()` and `writeAsCsv()`) do as the name suggests, the third one 
+can be used to specify a custom data output format. Please refer to [Data Sinks](#data-sinks) for 
+more information on writing to files and also about custom data output formats.
+
+The `print()` method is useful for developing/debugging. It will output the contents of the DataSet 
+to standard output (on the JVM starting the Flink execution). **NOTE** The behavior of the `print()`
+method changed with Flink 0.9.x. Before it was printing to the log file of the workers, now its 
+sending the DataSet results to the client and printing the results there.
+
+`collect()` allows to retrieve the DataSet from the cluster to the local JVM. The `collect()` method 
+will return a `List` containing the elements.
+
+Both `print()` and `collect()` will trigger the execution of the program.
```

Comment:
Let's be explicit. Something along the lines of: ...trigger execution of the program. You don't need a further call to `execute()`.